apps/go/manager/tests/circular_buffer_test.go

```diff
@@ -0,0 +1,207 @@
+package tests
+
+import (
+	"fmt"
+	"manager/types"
+	"time"
+)
+
+// define a test suite struct
+type CircularBuffertUnitTestSuite struct {
+	BaseSuite
+}
+
+// Size of the buffer to test
+const testBufferLen uint32 = 50
+
+func (s *CircularBuffertUnitTestSuite) Test_CircularBuffer() {
```

Comment:
Split this into different functions for each case like this:
- All Good
- Overflow
- Should fail
- etc